rules/0131.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rules
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/rules/descriptor"
+	p "google.golang.org/protobuf/reflect/protoreflect"
+)
+
+func init() {
+	registerRules(
+		checkRequestMessageName(),
+		checkRequestMessageNameField(),
+		checkRequestMessageUnknownFields(),
+		checkResponseMessageName(),
+	)
+}
+
+// Get messages should have a properly named Request message.
+func checkRequestMessageName() lint.Rule {
+	return &descriptor.CallbackRule{
+		RuleInfo: lint.RuleInfo{
+			Name:         lint.NewRuleName("core", "0131", "request_message", "name"),
+			Description:  "check that Get RPCs have appropriate request messages",
+			RequestTypes: []lint.RequestType{lint.ProtoRequest},
+			URI:          "https://aip.dev/131#guidance",
+		},
+		Callback: descriptor.Callbacks{
+			MethodCallback: func(m p.MethodDescriptor, s lint.DescriptorSource) (problems []lint.Problem, err error) {
+				// We only care about Get- methods for the purpose of this rule;
+				// ignore everything else.
+				methodName := string(m.Name())
+				if !strings.HasPrefix(methodName, "Get") {
```

Comment:
Should probably introduce a library that encapsulates this type of logic for example, "isGetMethod", "isUpdateMethod", or maybe "isStandardMethod", "isCustomMethod".  Dunno, thinking out loud.

Reply:
I went to write this, but it seems like I would be replacing...

```go
if !strings.HasPrefix(methodName, "Get") {
	return
}
```

with...

```go
if isGetMethod(m) {
	return
}
```

...which does not actually seem better (particularly since I need `methodName` for other reasons frequently).

100% willing to be told I am missing something; I am out of my element here.

Reply:
One argument for this might be exceptions. For example, @garrettjonesgoogle recommended using `GetEffective-` as a slightly different prefix, and we would want to exempt it. Same for `GetIamPolicy`.

Okay, I talked myself into it.

Reply:
Done. Thanks!